From: Don Dutile <ddutile@redhat.com>
Date: Fri, 15 May 2009 10:21:17 -0400
Subject: [pci] missed fix to pci_find_upstream_pcie_bridge
Message-id: 4A0D7A5D.1060705@redhat.com
O-Subject: [RHEL5.4 PATCH 2/3] Backport missed fix to pci_find_upstream_pcie_bridge
Bugzilla: 500901
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Justin M. Forbes <jforbes@redhat.com>
RH-Acked-by: Chris Wright <chrisw@redhat.com>

BZ 500901

Chris Wright found this missing patch from 2.6.30-rc2
for IOMMU support.  I missed checking drivers/pci/search.c
when updating the 2.6.29-rc2 code backport to 2.6.30-rc2.
Trivial fix & patch.

  drivers/pci/search.c |    2 +-
  1 files changed, 1 insertions(+), 1 deletions(-)

Please review & ack.
Note: this bz is a beta blocker for 5.4.

- Don

diff --git a/drivers/pci/search.c b/drivers/pci/search.c
index 55962e7..2b3906b 100644
--- a/drivers/pci/search.c
+++ b/drivers/pci/search.c
@@ -29,7 +29,7 @@ pci_find_upstream_pcie_bridge(struct pci_dev *pdev)
 	if (pdev->is_pcie)
 		return NULL;
 	while (1) {
-		if (!pdev->bus->self)
+		if (!pdev->bus->parent)
 			break;
 		pdev = pdev->bus->self;
 		/* a p2p bridge */
